### PR TITLE
fix (react/19/replace-default-props): fix tsconfig.json error and package-lock.json

### PR DESCRIPTION
**Pull request by amirabbas-gh**
#### 📚 Description

This PR addresses a critical error found in the `tsconfig.json` file and fixes several issues related to `package-lock.json`. The changes include:
1. Updating `tsconfig.json` with the correct configuration to resolve runtime errors.
2. Removing the outdated `package-lock.json` and reinstalling dependencies using `pnpm` for consistency.
3. Fixing errors in lines 11 and 12 of the codemod file, which were causing unexpected behavior during execution.

#### 🧪 Test Plan

1. Updated the `tsconfig.json` file and verified it by running the project to ensure the error no longer occurs.
2. Removed `package-lock.json`, reinstalled dependencies using `pnpm`, and tested the project setup from scratch to ensure compatibility.
3. Verified that lines 11 and 12 of the codemod file now execute without errors and deliver the expected output.

#### 📄 Documentation to Update

- Regenerate the`tsconfig.json` file.
- Remove the `package-lock.json` file.
- Install the packages using `pnpm`.
- Edit the Codemod file to import the `@codemod.com` dependency.

### CODE REVIEW

File: react/19/replace-default-props/pnpm-lock.yaml
Comment (mohebifar) [inline react/19/replace-default-props/pnpm-lock.yaml:1]: We don't want any lock files in the codemod directories. this is going to be a monorepo and there'll be one lock file in the root of the repo.

File: react/19/replace-default-props/tsconfig.json
Comment (mohebifar) [inline react/19/replace-default-props/tsconfig.json:1]: I would say tsconfig should also be shared across all workspaces. You can create a package in a directory named tooling like `tooling/tsconfig` and extend the config for all the codemods.

```
{
  "extends": "@tooling/tsconfig/codemod.json"
}
```

Comment (amirabbas-gh) [reply react/19/replace-default-props/tsconfig.json:1]: how should i do it?

Comment (mohebifar) [reply react/19/replace-default-props/tsconfig.json:1]: Use this as an example:

The tooling package just for the tsconfig:

https://github.com/t3-oss/create-t3-turbo/blob/main/tooling/typescript/base.json

Usage in packages that need tsconfig:

https://github.com/t3-oss/create-t3-turbo/blob/main/packages/api/tsconfig.json#L2

**Comment by amirabbas-gh** [comment]
@mohebifar I did all what you say.

**Review by mohebifar — APPROVED** [review]

**Pull request merged**